#!/usr/bin/env python
PACKAGE="vision_tasks"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("clahe_clip", double_t, 0, "CLAHE clip limit", 2.0, 0.0, 40.0);
gen.add("clahe_grid_size", int_t, 0, "CLAHE grid size", 3, 1, 16);
gen.add("clahe_bilateral_iter", int_t, 0, "Bilateral filter on CLAHE iterations", 0, 0, 16);
gen.add("balanced_bilateral_iter", int_t, 0, "Bilateral filter on white balanced iterations", 0, 0, 8);
gen.add("denoise_h", double_t, 0, "Denoising h value", 20.0, 0, 20.0);
gen.add("hough_minline", int_t, 0, "Minimum Length of line", 50, 0, 500);
gen.add("hough_threshold", int_t, 0, "Hough threshold", 50, 0, 500);
gen.add("hough_maxgap", int_t, 0, "Maximum gap in line", 50, 0, 500);
gen.add("canny_threshold_low", int_t, 0, "Lower bound of canny", 232, 0, 500);
gen.add("canny_threshold_high", int_t, 0, "Higher bound of canny", 250, 0, 500);
gen.add("low_b", int_t, 0, "Lower Bound Of B", 254, 0, 255);
gen.add("high_b", int_t, 0, "Higher Bound Of B", 255, 0, 255);
gen.add("low_g", int_t, 0, "Lower Bound Of G", 254, 0, 255);
gen.add("high_g", int_t, 0, "Higher Bound Of G", 255, 0, 255);
gen.add("low_r", int_t, 0, "Lower Bound Of R", 254, 0, 255);
gen.add("high_r", int_t, 0, "Higher Bound Of R", 255, 0, 255);
gen.add("closing_mat_point", int_t, 0, "", 1, 1, 7);
gen.add("closing_iter", int_t, 0, "Closing iterations", 1, 0, 10);
gen.add("filterByArea", bool_t, 0, "Filter by area", True)
gen.add("filterByCircularity", bool_t, 0, "Filter by circularity", True)
gen.add("filterByColor", bool_t, 0, "Filter by color", True)
gen.add("filterByConvexity", bool_t, 0, "Filter by convexity", False)
gen.add("filterByInertia", bool_t, 0, "Filter by intertia", False)
gen.add("minThreshold", double_t, 0, "Threshold min", 0.0, 0.0, 100.0)
gen.add("maxThreshold", double_t, 0, "Threshold max", 100.0, 0.0, 100.0)
gen.add("thresholdStep", double_t, 0, "Threshold step", 1.0, 0.1, 100.0)
gen.add("minDistBetweenBlobs", double_t, 0, "Minimum distance between two blobs", 1.0, 0.1, 10000000.0)
gen.add("minArea", double_t, 0, "Area min", 1000.0, 0.0, 100000000.0)
gen.add("maxArea", double_t, 0, "Area max", 100000000.0, 0.0, 100000000.0)
gen.add("minCircularity", double_t, 0, "Circularity min", 0.0, 0.0, 1.0)
gen.add("maxCircularity", double_t, 0, "Circularity max", 0.05, 0.0, 1.0)
gen.add("minConvexity", double_t, 0, "Convexity min", 0.0, 0.0, 1.0)
gen.add("maxConvexity", double_t, 0, "Convexity max", 1.0, 0.0, 1.0)
gen.add("minInertiaRatio", double_t, 0, "Inertia min", 0.0, 0.0, 1.0)
gen.add("maxInertiaRatio", double_t, 0, "Inertia max", 1.0, 0.0, 1.0)

exit(gen.generate(PACKAGE, "vision_tasks", "gateFrontRange"));
